packages/auto-id/src/utils.ts

```diff
@@ -1,28 +1,73 @@
+import { compactAddLength } from '@polkadot/util'
 import { ObjectIdentifier } from 'asn1js'
-import { randomBytes } from 'crypto'
 
 /**
- * Encodes a given string representation of an OID into its DER format.
- * This function is specifically used to encode signature algorithm OIDs.
+ * Encodes a given string representation of an OID into its DER format,
+ * and then applies SCALE encoding by prepending the length of the DER-encoded data in compact format.
  *
  * @param oid The string representation of the ObjectIdentifier to be encoded.
- * @returns Uint8Array containing the DER encoded OID along with NULL params of X.509 signature algorithm.
+ * @returns Uint8Array containing the SCALE encoded, DER-encoded OID with appended NULL parameter.
  */
 export function derEncodeSignatureAlgorithmOID(oid: string): Uint8Array {
   const objectIdentifier = new ObjectIdentifier({ value: oid })
-  const berArrayBuffer = objectIdentifier.toBER(false)
+  const oidEncoded = objectIdentifier.toBER(false)
 
   // Typically, in X.509, the algorithm identifier is followed by parameters; for many algorithms, this is just NULL.
   const nullParameter = [0x05, 0x00] // DER encoding for NULL
 
   // Calculate the total length including OID and NULL parameter
-  const totalLength = berArrayBuffer.byteLength + nullParameter.length
+  const totalLength = oidEncoded.byteLength + nullParameter.length
 
-  const sequenceHeader = [0x30, totalLength] // 0x30 is the DER tag for SEQUENCE
+  // Construct the sequence header
+  const sequenceHeader = [0x30] // 0x30 is the DER tag for SEQUENCE
+  if (totalLength < 128) {
+    sequenceHeader.push(totalLength) // Short form length
+  } else {
+    // Long form length encoding
+    const lengthBytes = []
+    let tempLength = totalLength
+    while (tempLength > 0) {
+      lengthBytes.push(tempLength & 0xff)
+      tempLength >>= 8
+    }
+    sequenceHeader.push(0x80 | lengthBytes.length, ...lengthBytes.reverse())
+  }
 
-  return new Uint8Array([...sequenceHeader, ...new Uint8Array(berArrayBuffer), ...nullParameter])
+  // Combine the sequence header, OID, and NULL parameter into one Uint8Array
+  const derSequence = new Uint8Array(
+    sequenceHeader.length + oidEncoded.byteLength + nullParameter.length,
+  )
+  derSequence.set(sequenceHeader, 0)
+  derSequence.set(new Uint8Array(oidEncoded), sequenceHeader.length)
+  derSequence.set(nullParameter, sequenceHeader.length + oidEncoded.byteLength)
+
+  // Apply SCALE encoding by prepending the compact length of the entire DER sequence
+  return compactAddLength(derSequence)
```

Comment:
Is applying SCALE encoding really necessary? I don't recall doing this in `auto-kit`.

Reply:
Yes in TS. Otherwise, it's not taking the argument properly.
I had created an issue on slack: https://subspacenetwork.slack.com/archives/C03QY5Z0USF/p1719385961803199 before using SCALE.

Reply:
The SCALE encoding should not happen here. If it is needed to submit the transaction, it should happen in the submission code. This should just return the DER encoded signature algorithm.

Reply:
I get this error, if i don't use SCALE encoding: `compactAddLength(derSequence)`.
```sh
Error: createType(Call):: Call: failed decoding autoId.registerAutoId:: Struct: failed on args: {"req":"{\"_enum\":{\"X509\":\"PalletAutoIdRegisterAutoIdX509\"}}"}:: Struct: failed on req: {"_enum":{"X509":"PalletAutoIdRegisterAutoIdX509"}}:: Enum(X509):: Enum(Root):: Struct: failed on signatureAlgorithm: Bytes:: Bytes: required length less than remainder, expected at least 13, found 9
at createTypeUnsafe (/Users/abhi3700/F/coding/github_repos/subspace/auto-sdk/node_modules/@polkadot/types-create/cjs/create/type.js:54:22)
    at TypeRegistry.createTypeUnsafe (/Users/abhi3700/F/coding/github_repos/subspace/auto-sdk/node_modules/@polkadot/types/cjs/create/registry.js:230:52)
    at extrinsicFn (/Users/abhi3700/F/coding/github_repos/subspace/auto-sdk/node_modules/@polkadot/types/cjs/metadata/decorate/extrinsics/createUnchecked.js:16:25)
    at Object.decorated [as registerAutoId] (/Users/abhi3700/F/coding/github_repos/subspace/auto-sdk/node_modules/@polkadot/api/cjs/base/Decorate.js:491:50)
    at receipt (/Users/abhi3700/F/coding/github_repos/subspace/auto-sdk/packages/auto-id/src/registry.ts:166:26)
    at new Promise (<anonymous>)
    at Registry.<anonymous> (/Users/abhi3700/F/coding/github_repos/subspace/auto-sdk/packages/auto-id/src/registry.ts:165:46)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/abhi3700/F/coding/github_repos/subspace/auto-sdk/packages/auto-id/src/registry.ts:28:58)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Reply:
Per the name (`derEncodeSignatureAlgorithmOID`) this function should return DER encoded data. I think you should SCALE encode it [here](https://github.com/subspace/auto-sdk/blob/e88852cc620da5c7b9058e591e1c66e38bb77855/packages/auto-id/src/registry.ts#L107))